sync-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -81,6 +93,16 @@ public HttpConnection(String requestMethod,
         this.url = url;
         this.contentType = contentType;
         this.requestProperties = new HashMap<String, String>();
+        this.requestFilters = new LinkedList<HttpConnectionRequestFilter>();
+        this.responseFilters = new LinkedList<HttpConnectionResponseFilter>();
+    }
+
+    /**
+     * Sets the number of times this request can be retried.
+     * @param numberOfRetries the number of times this request can be retried.
+     */
+    public void setNumberOfRetries(int numberOfRetries){
+        this.numberOfRetries = numberOfRetries;
```

Comment:
I'd interpret it as it being modified as we are using it, but thinking about it again, I don't think we need to do either, since the execute method takes a copy of the number of retries before executing the loop. So we could make a documentation change to say that the new value is ignored if called after execute has been called